src/components/cylc/Toolbar.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <v-toolbar
+    flat
+    dense
+    class="c-toolbar"
+  >
+    <div class="v-toolbar-title">
+      <v-toolbar-title
+        class="tertiary--text font-weight-light"
+      >
+        <!-- burger button for mobile -->
+        <v-btn
+          v-if="responsive"
+          class="default v-btn--simple"
+          dark
+          icon
+          @click.stop="onClickBtn"
+        >
+          <v-icon>mdi-view-list</v-icon>
+        </v-btn>
+        <!-- title -->
+        <span class="c-toolbar-title">{{ title }}</span>
+      </v-toolbar-title>
+    </div>
+
+    <!-- control bar elements displayed only when a suite has been positioned -->
+    <template v-if="tree && tree.length > 0">
+      <a @click="onClickPause">
+        <v-icon color="#5E5E5E">mdi-pause</v-icon>
+      </a>
+
+      <a @click="onClickStop">
+        <v-icon color="#5E5E5E">mdi-stop</v-icon>
+      </a>
+
+      <a>
+        <v-chip color="#E7E7E7" @click="toggleExtended">Control</v-chip>
+      </a>
+
+      <span>Running, will stop at 30000101T0000 cycle</span>
+
+      <v-spacer />
+
+      <a class="add-view" @click="onClickAddView">
+        Add View <v-icon color="#5995EB">mdi-plus-circle</v-icon>
+      </a>
+    </template>
+
+    <!-- displayed only when extended===true -->
+    <template v-slot:extension v-if="extended">
+      <span style="margin-left: 260px;">
+        <a @click="onClickPause">
+          <v-icon color="#5E5E5E">mdi-pause</v-icon>
+        </a>
+
+        <a @click="onClickStop">
+          <v-icon color="#5E5E5E">mdi-stop</v-icon>
+        </a>
+
+        <span>Other controls added in the future</span>
+      </span>
+    </template>
+
+  </v-toolbar>
+</template>
+
+<script>
+
+  import { mapMutations, mapState } from 'vuex'
+
+  export default {
+  data: () => ({
+    responsive: false,
+    responsiveInput: false,
+    extended: false
+  }),
+
+  computed: {
+    ...mapState('app', ['title']),
+    ...mapState('suites', ['tree'])
+  },
+
+  mounted () {
+    this.onResponsiveInverted()
+    window.addEventListener('resize', this.onResponsiveInverted)
+  },
+  beforeDestroy () {
+    window.removeEventListener('resize', this.onResponsiveInverted)
```

Comment:
Sure, but this came from the vuetify original code, not written by us... I assumed they had a reason... but didn't think too much about it

Reply:
If we need to do anything programmatic then JS is the way to go, otherwise CSS FTW!

tldr; I have a bit of trauma from a previous project:

I once built an unconventional responsive site which needed to respond to viewport aspect ratio (before CSS3 was sufficiently universal) and had a hell of a lot of problems with resize event listeners, some issues I had:

* Browsers trigger events differently on maximise.
* Mobile screen rotation.
* Callback delay can cause incorrect measurements especially in click+drag resizing.
* Fullscreen mode.
* Jerky resizing.

Reply:
>tldr; I have a bit of trauma from a previous project:

Gee! Definitely better if we can avoid repeating those mistakes. Doesn't seem simple to implement responsiveness in JS.

>If we need to do anything programmatic then JS is the way to go, otherwise CSS FTW!

Agreed. As this is code from Vuetify, I would prefer to tackle this one later. When doing so, might be worth to look at other parts of the code for similar pattern as well.

Reply:
Cool.